### PR TITLE
[Ripple] Additional snapshot test for an edge case

### PR DESCRIPTION
**Pull request by yarneo**
This snapshot test is added to test the behavior when the rippleHighlighted is set to YES and the ripple is then dissolved because there is no check in the implementation to only dissolve when rippleHighlighted is set to NO.

This is a pre-snapshot test to successfully test PR: #7274

### CODE REVIEW

File: components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m [hunk 1]
```diff
@@ -87,4 +87,19 @@ - (void)testDraggedRippleView {
...
+  [self.rippleView setRippleColor:UIColor.orangeColor forState:MDCRippleStateHighlighted];
+  self.rippleView.rippleHighlighted = YES;
+  [self.rippleView setValue:@YES forKey:@"_tapWentInsideOfBounds"];
+  [self.rippleView setValue:@YES forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = NO;
+  [self.rippleView setValue:@NO forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = YES;
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.view];
```
Comment (jverkoey) [inline components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m:103]: Will the animations have had a chance to run with the way this test is written? Core Animation animations are typically added at the turn of the run loop but I don't see a moment where we let the run loop run here.

Comment (yarneo) [reply components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m:103]: That's why I set the speed to 10000 for the ripple layer to allow the animations to run instantly.

Comment (yarneo) [reply components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m:103]: added a runloop run instead of a layer speed change.

Comment (jverkoey) [reply components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m:103]: To clarify: the speed won't take any effect until the run loop has had a chance to commit the animations to the render server, so a run loop pump (or a `[CATransaction flush]`) is necessary for the animations to take effect.

File: components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m [hunk 2]
```diff
@@ -87,4 +87,19 @@ - (void)testDraggedRippleView {
...
+- (void)testRippleHighlightedIsNotDissolvedWhenSetToYES {
+  // Given
+  [self.rippleView setRippleColor:UIColor.orangeColor forState:MDCRippleStateHighlighted];
+  self.rippleView.rippleHighlighted = YES;
+  [self.rippleView setValue:@YES forKey:@"_tapWentInsideOfBounds"];
+  [self.rippleView setValue:@YES forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = NO;
+  [self.rippleView setValue:@NO forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = YES;
+  [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:2.0]];
```
Comment (jverkoey) [inline components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m]: In the interest of speed a combination of what you had before (ramped up layer speed) + a low time interval here should be sufficient to ensure that the animation has taken effect.

**Review by jverkoey — APPROVED** [review]

File: components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m (hunk 1 above)
Comment (yarneo) [reply components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m:103]: Done! thanks

File: components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m [hunk 3]
```diff
@@ -87,4 +87,20 @@ - (void)testDraggedRippleView {
...
+  // Given
+  self.rippleView.layer.speed = 10000;
+  [self.rippleView setRippleColor:UIColor.orangeColor forState:MDCRippleStateHighlighted];
+  self.rippleView.rippleHighlighted = YES;
+  [self.rippleView setValue:@YES forKey:@"_tapWentInsideOfBounds"];
+  [self.rippleView setValue:@YES forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = NO;
+  [self.rippleView setValue:@NO forKey:@"_tapWentOutsideOfBounds"];
+  self.rippleView.rippleHighlighted = YES;
+  [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:0.5]];
```
Comment (jverkoey) [inline components/Ripple/tests/snapshot/MDCStatefulRippleViewSnapshotTests.m:100]: 0.5 is still an eon when things are moving at 10000x :P
